pkg/plugins/pg/source/cdc.go

```diff
@@ -132,6 +147,8 @@ func (s *Source) withCDC(ctx context.Context, cfg plugins.Config) error {
 	go func() {
```

Comment:
I noticed one thing in the code which has been merged previously:
```go
if err != nil {
	if !strings.Contains(err.Error(), "SQLSTATE 42710") {
		return cerrors.Errorf("failed to create publication %s: %w", slotName, err)
	}
	log.Printf("publication %s already exists; continuing to slot setup", publication)
}
```
My understanding is that we want all DML operations to be published by this this publication. However, if the publication already exists, it _may_ be set up such that only some DML operations are being published. E.g. the publication might already exist, but only `insert`s are published.

Reply:
This is probably something we need to point out in the documentation. It can also exist if we've created it before, which is why we have to detect that error manually like this. I agree, though, that could be a foot-gun to the end user that we can't really stop. Our best advice is probably to recommend a unique name for the connector's slot in all cases unless you specifically know what you're doing reusing an established slot.

Reply:
So, I haven't used Pg publications before, but if we start using an existing publication, will we get all operations from it, or just those from the moment we subscribe to it?

One thing we can do, to ensure that the publication we use is publishing all the operations, is to read its metadata. If all the operations are there, we should be fine. If not, we may even want to fail or create a new one (if it won't consume too many resources).

Reply:
When you start using an existing replication slot, you specify in the pglogrepl subscription where you want to start in the WAL log. Currently it always starts at zero, meaning it will read all of the contents of the log out. This will be changed to be passed to a specific LSN soon. We get access to a set of [pglogrepl message types](https://github.com/jackc/pglogrepl/blob/master/message.go), but I'm not sure how we could reliably differentiate between the absence of an event type and regular operation if we encounter no other errors.

Reply:
For the rest: we talked about this privately. Checking how an existing publication has been setup can be done in a separate PR.

Reply:
> Currently it always starts at zero, meaning it will read all of the contents of the log out. This will be changed to be passed to a specific LSN soon.

@dylanlott Is there a follow-up issue for that? If not we should create one.